discussions.go

```diff
@@ -0,0 +1,1027 @@
+
+	req, err := s.client.NewRequest("POST", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	d := new(Discussion)
+	resp, err := s.client.Do(req, d)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return d, resp, err
+}
+
+// UpdateMergeRequestDiscussionNoteOptions represents the available
+// UpdateMergeRequestDiscussion() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/discussions.html#modify-existing-merge-request-discussion-note
+type UpdateMergeRequestDiscussionNoteOptions struct {
+	Body      *string    `url:"body,omitempty" json:"body,omitempty"`
+	CreatedAt *time.Time `json:"created_at,omitempty"`
+}
+
+// UpdateMergeRequestDiscussionNote modifies existing discussion of a merge
+// request.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/discussions.html#modify-existing-merge-request-discussion-note
+func (s *DiscussionsService) UpdateMergeRequestDiscussionNote(pid interface{}, mergeRequest int, discussion string, note int, opt *UpdateMergeRequestDiscussionNoteOptions, options ...OptionFunc) (*Discussion, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/merge_requests/%d/discussions/%s/notes/%d",
+		url.QueryEscape(project), mergeRequest, discussion, note)
+
+	req, err := s.client.NewRequest("PUT", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	d := new(Discussion)
+	resp, err := s.client.Do(req, d)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return d, resp, err
+}
+
+// DeleteMergeRequestDiscussionNote deletes an existing discussion of a merge
+// request.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/discussions.html#delete-a-merge-request-discussion-note
+func (s *DiscussionsService) DeleteMergeRequestDiscussionNote(pid interface{}, mergeRequest int, discussion string, note int, options ...OptionFunc) (*Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, err
+	}
+	u := fmt.Sprintf("projects/%s/merge_requests/%d/discussions/%s/notes/%d",
+		url.QueryEscape(project), mergeRequest, discussion, note)
+
+	req, err := s.client.NewRequest("DELETE", u, nil, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Do(req, nil)
+}
+
+// ListProjectCommitDiscussionsOptions represents the available
+// ListCommitDiscussions() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/discussions.html#list-project-commit-discussions
+type ListProjectCommitDiscussionsOptions struct {
+	ListOptions
+}
+
+// ListProjectCommitDiscussions gets a list of all discussions for a single
+// commit.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/discussions.html#list-project-commit-discussions
+func (s *DiscussionsService) ListProjectCommitDiscussions(pid interface{}, commit string, opt *ListProjectCommitDiscussionsOptions, options ...OptionFunc) ([]*Discussion, *Response, error) {
```

Comment:
The docs are unfortunately incorrect, as commits cannot be identified by ints. It looks to me like a bug in GitLab, since they ordinarily return the int ID of the object the note applies to, but for commit discussion notes, the fields `noteable_id` and `noteable_iid` are `null` (in the JSON response payload).

Reply:
Check! 👍